assets/src/functions/deepClone.js

```diff
@@ -0,0 +1,59 @@
+// Following is needed to get the right global object in all environments.
+// See https://stackoverflow.com/a/6930376/4961158.
+let global;
+try {
+  global = Function('return this')();
+} catch(e) {
+  global = window;
+}
+
+// A deep clone that has the same result as doing JSON.parse(JSON.stringify(value)).
+// It's 2 to 5 times faster than both JSON switcheroo and lodash's cloneDeep in most cases.
+// Only for really large object sizes, e.g. 500KB and over it's roughly the same or a little slower than Lodash.
+// It does not preserve the object in the same way as Lodash does, i.e. non-built in classes are not preserved, just
+// like they aren't in the JSON switcheroo approach. So it's only intended to be used where you would otherwise be able
+// to use the JSON approach.
+const _deepClone = (value, ancestors = [], clones = []) => {
+  const type = typeof value;
+  if (type === 'function') {
+    return undefined;
+  }
+  if (type !== 'object') {
+    return value;
+  }
+  if (ancestors.includes(value)) {
+
+    return clones[ancestors.indexOf(value)];
```

Comment:
I made this one return the already cloned reference, if there's a circular reference. But maybe it makes more sense to throw an exception here, as that would also happen when doing `JSON.stringify(objectWithCircularReference)`, and circular references should in practice not happen unless we made a mistake somewhere.